### PR TITLE
tests: Move from t.Errorf and others.

### LINKED ISSUE #8063: tests: Move from `t.Errorf(` `t.Fail` usages on comparison to `testutil.Ok` / `Assert` / `Equals`
Example (https://github.com/prometheus/prometheus/blob/a282d2509971174301408c5a5f96946c478dfb0f/tsdb/querier_test.go#L1526):

```
                       if len(matches) != len(c.exp) {
				t.Errorf("Evaluating %s, length of result not equal to exp", c.pattern)
			} else {
				for i := 0; i < len(c.exp); i++ {
					if c.exp[i] != matches[i] {
						t.Errorf("Evaluating %s, unexpected result %s", c.pattern, matches[i])
					}
				}
			}
```
Should be  just:
```
testutil.Equals(t, exp, matches)
```

Potential good first issue for @GayathriVenkatesh

### PR DESCRIPTION
**Pull request by Creatone**
Fixes #8063.

### CODE REVIEW

**Review by roidelapluie — APPROVED** [review]
Thanks, this must have taken a lot of time!

I have added a few comments.

May I recommend you to split this into multiple pull requests? Maybe 4-5 small packages per pull request (files) ? It will be easier to review and we can merge them incrementally.

**Review by roidelapluie — CHANGES_REQUESTED** [review]
Sorry, I did not want to mark this as approved yet.

**Comment by Creatone** [comment]
@roidelapluie No problem :)

**Comment by Creatone** [comment]
@roidelapluie I'm sorry for the force push, I forgot to sign revert commits.

**Comment by Creatone** [comment]
@roidelapluie Hope this split is enough.

File: web/api/v1/api_test.go
Comment (Creatone) [reply web/api/v1/api_test.go:233]: Because `ar.AlertingRules()` needs it.

**Comment by beorn7** [comment]
Running into this during our bug scrub. @roidelapluie it looks like your comments were all addressed, but you never commented on this again. Are you fine with it in general? If yes, then this needs a rebase. @Creatone would you still be willing to do that? Sorry for the long radio silence.

**Comment by bboreham** [comment]
No update for 6 months; I will assign to myself and try to bring it up to date.

**Comment by bboreham** [comment]
I rebased, squashed reverts and corrections, fixed up all the tests to pass, and updated a number of cases which are now flagged by the testify linter.
I also made a few simplifications to match what I think was the original intent of the test.

Personally I would not bother updating a plain `t.Fatal()` to `require.Fail()`, since the benefits are very slim, but I left those that were in this PR.

**Comment by beorn7** [comment]
@bboreham do you want another Prometheus maintainer to review this? If you feel comfortable with the current state, I would say just merge it. But happy to do another pass over it if you prefer.

**Comment by bboreham** [comment]
Yes, it's possible I mangled something and I am not the best person to detect that. 
However I am ok to merge it now.  I will need to dismiss @roidelapluie's "requested changes" review.

**Pull request merged**
